### PR TITLE
fix(lndhub): expired alby token #960

### PR DESCRIPTION
**Pull request by escapedcat**
### Describe the changes you have made in this PR

Takes care of expired Alby tokens

### Link this PR to an issue

#960

### Type of change (Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

Using https://www.charlesproxy.com  
Old school, but worked :D

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

File: src/extension/background-script/connectors/lndhub.ts [hunk 1]
```diff
@@ -353,15 +352,31 @@ export default class LndHub implements Connector {
...
+
         if (errResponse?.status === 404) {
           const method = path.replace("/", "");
           throw new Error(`${method} not supported by the connected account.`);
         }
+
+        if (errResponse?.status === 401) {
+          console.warn(
+            "TOKEN MIGHT BE EXPIERED, LET'S TRY TO GET YOU LOGGED IN"
+          );
```
Comment (bumi) [inline src/extension/background-script/connectors/lndhub.ts:364]:
```suggestion
```

as this is a normal case I would not log anything here. (we also did not use all-caps messages so far).
if we want to log something we could do a console.info() in the authorize() function?

File: src/extension/background-script/connectors/lndhub.ts [hunk 2]
```diff
@@ -375,9 +390,11 @@ export default class LndHub implements Connector {
         throw new Error(data.message);
       }
     }
+
     if (defaultValues) {
-      data = Object.assign(Object.assign({}, defaultValues), data);
```
Comment (bumi) [inline src/extension/background-script/connectors/lndhub.ts:379]: those default values are left-overs from joule, there default values had been used. we don't really do that I think.

**Review by bumi — APPROVED** [review]
utACK.

File: src/extension/background-script/connectors/lndhub.ts (hunk 1 above)
Comment (escapedcat) [reply src/extension/background-script/connectors/lndhub.ts]: Removed

File: src/extension/background-script/connectors/lndhub.ts (hunk 2 above)
Comment (escapedcat) [reply src/extension/background-script/connectors/lndhub.ts:379]: Removed it all connectors where it's not used. [Only connector were it's used is lnd](https://github.com/getAlby/lightning-browser-extension/blob/fix/960_expired-lndhub-alby-token/src/extension/background-script/connectors/lnd.ts/#L199-L205).

**Pull request merged**